packages/core-components/src/components/DependencyGraph/DependencyGraph.tsx

```diff
@@ -431,28 +444,43 @@ export function DependencyGraph<NodeData, EdgeData>(
     updateGraph,
   ]);
 
-  function setNode(id: string, node: Types.DependencyNode<NodeData>) {
-    graph.current.setNode(id, node);
-    updateGraph();
-    return graph.current;
-  }
+  const setNode = useCallback(
+    (id: string, node: Types.DependencyNode<NodeData>) => {
+      graph.current.setNode(id, node);
+      updateGraph();
+      return graph.current;
+    },
+    [updateGraph],
+  );
 
-  function setEdge(id: dagre.Edge, edge: Types.DependencyEdge<EdgeData>) {
-    graph.current.setEdge(id, edge);
-    updateGraph();
-    return graph.current;
-  }
+  const setEdge = useCallback(
+    (id: dagre.Edge, edge: Types.DependencyEdge<EdgeData>) => {
+      graph.current.setEdge(id, edge);
+      updateGraph();
+      return graph.current;
+    },
+    [updateGraph],
+  );
 
   return (
-    <div ref={containerRef} className={styles.root}>
+    <div
+      ref={containerRef}
+      className={combineClasses(
+        styles.root,
+        fit === 'contain' && styles.fixedHeight,
+      )}
+    >
       <FullScreen
         handle={fullScreenHandle}
-        className={fullScreenHandle.active ? styles.fullscreen : styles.root}
+        className={combineClasses(
```

Comment:
Maybe use `classnames` utility like in other components instead of introducing own function?

Reply:
yeah agreed